misc/phase_retrieval_tomopy_comparison.ipynb

```diff
@@ -0,0 +1,1264 @@
+    "ax.plot(y_range, recon_attenuation.array[y_range,x_slice])\n",
+    "ax.plot(y_range, recon_filter.array[y_range,x_slice])\n",
+    "\n",
+    "ax.set_xlabel('Horizontal y')\n",
+    "ax.set_ylabel('Intensity')\n",
+    "ax.set_title('Line Profile at horizontal_x=' + str(x_slice) + ', vertical slice=' + str(vertical_slice))\n",
+    "ax.legend(['Original', 'Phase retrieval - scaled by mu', 'Phase retrieval - full_retrieval=False'])\n",
+    "\n",
+    "plt.tight_layout()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that both methods blur the original image. The phase retrieval method becomes negative because of the values > 1 are passed into the negative log. This may be an indication that the full phase retrieval is not valid for this experimental setup, in which case using `full_retrieval = False` may be more useful as it just applies a Paganin-like filter to the data "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Compare the signal to noise ratio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Original reconstruction SNR ={:.3f}\".format(np.abs(recon.mean()/recon.array.std())))\n",
+    "print(\"Phase retrieved reconstruction SNR = {:.3f}\".format(np.abs(recon_attenuation.mean()/recon_attenuation.array.std())))\n",
+    "print(\"Phase retrieved (full_retrieval=False) reconstruction SNR = {:.3f}\".format(np.abs(recon_filter.mean()/recon_filter.array.std())))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The negative values in the phase retrival skew the result but with full_retrieval=False the SNR is improved"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Generalised Paganin method"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The generalised Paganin method is implemented in CIL following the description in https://iopscience.iop.org/article/10.1088/2040-8986/abbab9 \\\n",
+    "When features in the image are close to the Nyquist frequency of the system, a more generalised form of the Pagnin filter can be used which preserves these high frequency features while still boosting SNR. This may have a similar effect to applying an unsharp mask after the normal Paganin phase retrieval. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "$$\n",
+    "T(x,y) =  -\\frac{1}{\\mu}\\ln\\left (\\mathcal{F}^{-1}\\left (\\frac{\n",
+    "        \\mathcal{F}\\left ( M^2I_{norm}(x, y,z = \\Delta) \\right )}{1 - \\frac{2\n",
+    "        \\alpha}{W^2}\\left ( \\cos(Wk_x) + \\cos(Wk_y) -2 \\right )}  \\right )\n",
+    "        \\right )\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Choose delta and beta"
```

Comment:
Why are these different to before? Can you motivate this choice?

Reply:
Good spot, they shouldn't be different. I've updated this now